### PR TITLE
WIP : Add target change

### PR DESCRIPTION
**Pull request by btry**
Updated work from goshiz's PR #332

### CODE REVIEW

File: inc/target.class.php [hunk 1]
```diff
@@ -56,15 +56,30 @@ public static function displayTabContentForItem(CommonGLPI $item, $tabnum = 1, $
       foreach ($found_targets as $target) {
          $i++;
          echo '<tr class="line'.($i % 2).'">';
+         switch ($target['itemtype']) {
+            case 'PluginFormcreatorTargetChange':
+               echo '<td onclick="document.location=\'../front/targetchange.form.php?id=' . $target['items_id'] . '\'" style="cursor: pointer">';
+               break;
+            case 'PluginFormcreatorTargetTicket':
+               echo '<td onclick="document.location=\'../front/targetticket.form.php?id=' . $target['items_id'] . '\'" style="cursor: pointer">';
+               break;
```
Comment (orthagh) [inline inc/target.class.php]: Could this part be factorized ? (Toolbox::getItemTypeFormURL)

File: inc/target.class.php [hunk 2]
```diff
@@ -56,15 +56,30 @@ public static function displayTabContentForItem(CommonGLPI $item, $tabnum = 1, $
       foreach ($found_targets as $target) {
          $i++;
          echo '<tr class="line'.($i % 2).'">';
+         switch ($target['itemtype']) {
```
Comment (orthagh) [inline inc/target.class.php]: Toolbox::getItemTypeFormURL

File: inc/target.class.php [hunk 3]
```diff
@@ -56,15 +56,30 @@ public static function displayTabContentForItem(CommonGLPI $item, $tabnum = 1, $
...
          echo $target['name'];
          echo '</td>';
 
          echo '<td align="center" width="32">';
          echo '<img src="'.$CFG_GLPI['root_doc'].'/plugins/formcreator/pics/pencil.png"
-                  alt="*" title="'.__('Edit').'"
-                  onclick="document.location=\'../front/targetticket.form.php?id='.$target['items_id'].'\'" align="absmiddle" style="cursor: pointer" /> ';
+                  alt="*" title="'.__('Edit');
+         switch ($target['itemtype']) {
+            case 'PluginFormcreatorTargetChange':
```
Comment (orthagh) [inline inc/target.class.php]: same for Toolbox::getItemTypeFormURL

File: inc/targetchange.class.php [hunk 4]
```diff
@@ -0,0 +1,1502 @@
+<?php
+class PluginFormcreatorTargetChange extends CommonDBTM
+{
+
+   static function getEnumDestinationEntity() {
```
Comment (orthagh) [inline inc/targetchange.class.php]: I think a super class who define theses enum functions could be ok.

Comment (btry) [inline inc/targetchange.class.php]: I was planning this an an other PR with other code refactors, to make things easier to  review.

**Pull request merged**